signal/signal.go

```diff
@@ -14,29 +14,40 @@ import (
 )
 
 var (
-	// interruptChannel is used to receive SIGINT (Ctrl+C) signals.
-	interruptChannel = make(chan os.Signal, 1)
-
-	// shutdownRequestChannel is used to request the daemon to shutdown
-	// gracefully, similar to when receiving SIGINT.
-	shutdownRequestChannel = make(chan struct{})
-
 	// started indicates whether we have started our main interrupt handler.
 	// This field should be used atomically.
 	started int32
+)
 
-	// quit is closed when instructing the main interrupt handler to exit.
-	quit = make(chan struct{})
+// Interceptor contains channels and methods regarding application shutdown
+// and interrupt signals
+type Interceptor struct {
+	// interruptChannel is used to receive SIGINT (Ctrl+C) signals.
+	interruptChannel chan os.Signal
 
 	// shutdownChannel is closed once the main interrupt handler exits.
-	shutdownChannel = make(chan struct{})
-)
+	shutdownChannel chan struct{}
+
+	// shutdownRequestChannel is used to request the daemon to shutdown
+	// gracefully, similar to when receiving SIGINT.
+	shutdownRequestChannel chan struct{}
+
+	// quit is closed when instructing the main interrupt handler to exit.
+	quit chan struct{}
+}
 
 // Intercept starts the interception of interrupt signals. Note that this
 // function can only be called once.
-func Intercept() error {
+func Intercept() (Interceptor, error) {
```

Comment:
nit: godoc above is now slightly outdated. Should say something like "Note that any previous active interceptor must be stopped before a new one can be created".

Reply:
Fixed.